demos/common/cpp/models/include/models/internal_model_data.h

```diff
@@ -38,23 +38,12 @@ struct InternalImageModelData : public InternalModelData
     int inputImgHeight;
 };
 
-struct InternalImageMatModelData : public InternalImageModelData
+struct InternalScaleData : public InternalImageModelData
 {
-    InternalImageMatModelData(const cv::Mat& mat) :
-        InternalImageModelData(mat.cols, mat.rows), mat(mat) {}
+    InternalScaleData(int width, int height, float scaleX, float scaleY) :
+        InternalImageModelData(width, height),
+        scaleX(scaleX), scaleY(scaleY) {}
 
-    InternalImageMatModelData(const cv::Mat& mat, int width, int height) :
-        InternalImageModelData(width, height), mat(mat) {}
-
-    cv::Mat mat;
-};
-
-struct InternalScaleMatData : public InternalModelData
-{
-    InternalScaleMatData(float scaleX, float scaleY, cv::Mat&& mat) :
-        x(scaleX), y(scaleY), mat(std::move(mat)) {}
-
-    float x;
-    float y;
-    cv::Mat mat;
+    float scaleX;
```

Comment:
names causing code like
`    const auto& scale = infResult.internalModelData->asRef<InternalScaleData>();
    float scaleX = stride / upsampleRatio * scale.scaleX;
    float scaleY = stride / upsampleRatio * scale.scaleY;`
could not it be something like
`const auto& scale_factors = infResult.internalModelData->asRef<InternalScaleData>();` ?

Reply:
Frankly we probably need to refactor this code and store actual sizes instead of scales. But this is feature for separate PR